buildbot/master.cfg

```diff
@@ -347,6 +347,145 @@ def make_fifoci_win(type, mode="normal"):
+                           descriptionDone="sign dmg",
+                           haltOnFailure=True))
+
+    if mode == "normal":
+        master_filename = WithProperties("/srv/http/dl/builds/dolphin-%s-%s.dmg", "branchname", "shortrev")
+        url = WithProperties("https://dl.dolphin-emu.org/builds/dolphin-%s-%s.dmg", "branchname", "shortrev")
+    elif mode == "wip":
+        master_filename = WithProperties("/srv/http/dl/wips/%s-dolphin-%s-%s.dmg", "author", "branchname", "shortrev")
+        url = WithProperties("https://dl.dolphin-emu.org/wips/%s-dolphin-%s-%s.dmg", "author", "branchname", "shortrev")
+    elif mode == "pr":
+        master_filename = WithProperties("/srv/http/dl/prs/%s-dolphin-latest.dmg", "branchname")
+        url = WithProperties("https://dl.dolphin-emu.org/prs/%s-dolphin-latest.dmg", "branchname")
+    else:
+        master_filename = url = ""
+
+    master_filename, url = map(get_sharded_dl_path, (master_filename, url))
+
+    if master_filename and url:
+         f.addSteps(ReliableFileUpload(workersrc="build/dolphin.dmg",
+                                      masterdest=master_filename,
+                                      url=url, keepstamp=True, mode=0o644))
+
+    if mode == "normal":
+
+         f.addStep(MasterShellCommand(command="/home/buildbot/bin/send_build.py",
+                                        env={
+                                            "BRANCH": WithProperties("%s", "branchname"),
+                                            "SHORTREV": WithProperties("%s", "shortrev"),
+                                            "HASH": WithProperties("%s", "revision"),
+                                            "AUTHOR": WithProperties("%s", "author"),
+                                            "DESCRIPTION": WithProperties("%s", "description"),
+                                            "TARGET_SYSTEM": "macOS",
+                                            "USER_OS_MATCHER": "osx",
+                                            "BUILD_URL": url,
+                                        },
+                                        description="notifying website",
+                                        descriptionDone="website notice"))
+
+         tmp_filename = WithProperties("/tmp/%s.7z", "revision")
+
+         f.addStep(MasterShellCommand(command=["/home/buildbot/bin/repack_dmg.sh", master_filename, tmp_filename],
+                                      description="converting dmg",
+                                      descriptionDone="convert dmg",
+                                      hideStepIf=StepWasSuccessful))
+
+         f.addStep(MasterShellCommand(command=["/home/buildbot/venv/bin/python", "/home/buildbot/bin/make_manifest.py",
+                                             "--input", tmp_filename,
+                                             "--version_hash", WithProperties("%s", "revision"),
+                                             "--platform", "macos",
```

Comment:
I would leave out auto-updates or put them on a different platform until we figure out how we want to handle these.

As it is we'll have osx-amd64 and osx-universal fighting for "what's the latest build on the macos platform dev branch".

Reply:
Sure, I'll update the platform to macos-universal so they don't conflict

Reply:
@delroth Does this send build need to be updated as well? 
```
f.addStep(MasterShellCommand(command="/home/buildbot/bin/send_build.py",
                                        env={
                                            "BRANCH": WithProperties("%s", "branchname"),
                                            "SHORTREV": WithProperties("%s", "shortrev"),
                                            "HASH": WithProperties("%s", "revision"),
                                            "AUTHOR": WithProperties("%s", "author"),
                                            "DESCRIPTION": WithProperties("%s", "description"),
                                            "TARGET_SYSTEM": "macOS",
                                            "USER_OS_MATCHER": "osx",
                                            "BUILD_URL": url,
                                        },
                                        description="notifying website",
                                        descriptionDone="website notice"))
```